### DISCUSSION TITLE
Reading a data from csv

### DISCUSSION THREAD

**Discussion by ramesh-khatri** [getgauge/taiko#1784] [Development]
Can anyone help me? I have done the following to get the data from the csv file

```
step("Enter the details in form", async () => {
    var csvData=[];
    fs.createReadStream('data\\pim\\employee.csv').pipe(csv()).on('data', function (row) {
        csvData.push(row)
    }).on('end',function() {
        console.log(csvData);   
        var firstName = csvData[0].first_name
        console.log(firstName)     // **It is printing the correct data**
        write(firstName, into(textBox({id:"firstName"}))); //**It is not writing the data as printed on firstName, It is writing some type of random name**
    })
});
```
I cannot use await before write(firstName, into ......) . Why it is not writing the valid data in id:firstName?

**Comment by zabil** [comment]
> I cannot use await before write

You must use `await` before write otherwise the call is returns a promise which explains why it's not writing the right value.
If you are using Gauge please pass the csv as a file parameter instead https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#special-parameter-csv

**Reply by ramesh-khatri** [reply]
Thank you so much @zabil .

**Reply by ramesh-khatri** [reply]
@zabil  Is there any example available for using csv as file parameter? (how to keep it on both spec and test file). I want to read a data from the csv and used that data to fill the form.

**Reply by zabil** [reply]
Please read https://docs.gauge.org/writing-specifications.html?os=macos&language=javascript&ide=vscode#step-with-table
